src/java/org/apache/cassandra/index/sai/disk/vector/OnDiskOrdinalsMap.java

```diff
@@ -231,17 +233,42 @@ private class FileReadingOrdinalsView implements OrdinalsView
     {
         RandomAccessReader reader = fh.createReader();
         private final long high = (segmentEnd - 8 - rowOrdinalOffset) / 8;
+        private final AtomicInteger lastFoundRowId = new AtomicInteger(-1);
+        private final AtomicLong lastFoundRowIdIndex = new AtomicLong(-1);
 
         /**
          * @return order if given row id is found; otherwise return -1
          */
         @Override
         public int getOrdinalForRowId(int rowId) throws IOException
         {
+            long low = 0;
+            if (lastFoundRowId.get() > -1 && lastFoundRowId.get() <= rowId)
+            {
+                low = lastFoundRowIdIndex.get();
+
+                // sequential read, skip binary search
+                if (low + 1 == rowId)
+                {
+                    long offset = rowOrdinalOffset + (lastFoundRowIdIndex.get() + 1) * 8;
+                    reader.seek(offset);
+                    int foundRowId = reader.readInt();
+                    if (foundRowId == rowId)
+                    {
+                        lastFoundRowId.incrementAndGet();
+                        lastFoundRowIdIndex.incrementAndGet();
```

Comment:
Can we set these regardless of `foundRowId == rowId`? I think we want to.

Reply:
no, we'll skip over some not yet read data in this case.

Reply:
I am not seeing when we'll skip data. Once we read `foundRowId`, we know that is the last row id, and we won't find any other row ids less than it because they are sorted in ascending order. Do you have an example to show that we'll skip data?

Reply:
with support for "lastFoundRowId == rowId" reads it won't skip